src/frontend/src/components/forms/ApiForm.tsx

```diff
@@ -139,6 +139,12 @@ export function OptionsApiForm({
   const formProps: ApiFormProps = useMemo(() => {
     const _props = { ...props };
 
+    // This forcefully overrides initial data
+    // Currently, most modals do not get pre-loaded correctly
```

Comment:
@XanderLuciano what do you mean by that? Do you have a sample?

Reply:
![image](https://github.com/inventree/InvenTree/assets/13877593/817bf954-ec98-4f81-ad78-d6be0dc7802c)
![image](https://github.com/inventree/InvenTree/assets/13877593/90752087-b760-4aa5-b5c2-5de15aa14fdf)
![image](https://github.com/inventree/InvenTree/assets/13877593/ffd84e85-8779-43ad-91a3-25b407418376)

If you look at the difference in the values for the `fields` object, there is a note able difference, and some of that data is required for the ApiFormField component to correctly appear, otherwise it shows the error message in the first photo. After the API call is made to the endpoint using OPTIONS and gets all the values for each field, it renders correctly / normally.

Reply:
I think it's primarily missing the `type` attribute, which would need to be added everywhere a model form is called. 

e.g. here 
![image](https://github.com/inventree/InvenTree/assets/13877593/55817659-d740-4fb4-9c05-9b01101ae9b1)

but that I think should be scoped out to a new issue.

Reply:
The idea behind that is that one can define which fields to show and which to hide. Only fields added through `filename: { /* ... */}` should be shown. And then one could add some customization to the defaults requested via the OPTIONS request. E.g. hook up a value state or make the disabled state react to some other condition.

Does this still work?

Reply:
@wolflu05 oh! Huh, surprisingly it does actually still work! Here is me removing the first name field on the edit user form
![image](https://github.com/inventree/InvenTree/assets/13877593/1d70da0d-4a92-4533-80ac-63150463674c)
![image](https://github.com/inventree/InvenTree/assets/13877593/40c8e622-5170-4ef7-9953-eebb82f81f05)

Reply:
ok, got it, you only override if data is not defined. I now also tested if the reactiveness (hooking up a state and use that for the disabled state (see playground)) still works, seems so.

Reply:
Yes! I re-used the idea of not rendering the entire ApiForm component if data was not available. Instead of not rending the component, I just override the fields temporarily until the data arrives. 

Now that I know the intended behavior, the method I chose actually seems quite fitting IMO.

Accidentally worked just right 😄 that's rare!

Reply:
@XanderLuciano @wolflu05 after testing the new code a bit more, I have found that this change breaks loading an "edit" form. The "initialData" query often returns first, before the "OPTIONS" (which tells us which fields are available).

Commenting out this line: `_props.fields = undefined` seems to result in correct behaviour again.

We need to review this ASAP as edit forms are really broken now